core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.js

```diff
@@ -0,0 +1,162 @@
+            errorCallback(errorResponse.data);
+          }
+        });
+      };
+
+      var _deleteExploration = function(
+          explorationId, role, successCallback, errorCallback) {
+        var editableExplorationDataUrl = (
+          UrlInterpolationService.interpolateUrl(
+          EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+            exploration_id: explorationId
+          })
+        );
+
+        params = {};
+        if (role) {
+          params.role = JSON.stringify(role);
+        }
+
+        $http.delete(editableExplorationDataUrl, {
+          params: params
+        }).then(function(response) {
+          // What does this return? What happens on Fail?
+          // What happens wtih ReadOnlyExplorationBackendApiService
+          // on success or fail.
+
+          // Delete item from the ReadOnlyExplorationBackendApiService's cache
+          ReadOnlyExplorationBackendApiService.deleteExplorationFromCache(
+            explorationId);
+
+          if (successCallback) {
+            successCallback(response);
+          }
+        }, function(errorResponse) {
+          if (errorCallback) {
+            errorCallback(errorResponse.data);
+          }
+        });
+      };
+
+      return {
+        fetchExploration: function(explorationId, applyDraft, v) {
+          return $q(function(resolve, reject) {
+            _fetchExploration(explorationId, applyDraft, v, resolve, reject);
+          });
+        },
+
+        /**
+         * Updates a exploration in the backend with the provided exploration
+         * ID. The changes only apply to the exploration of the given version
+         * and the request to update the exploration will fail if the provided
+         * exploration version is older than the current version stored in the
+         * backend. Both the changes and the message to associate with those
+         * changes are used to commit a change to the exploration.
+         * The new exploration is passed to the success callback,
+         * if one is provided to the returned promise object. Errors are passed
+         * to the error callback, if one is provided. Finally, if the update is
+         * successful, the returned exploration will be cached within the
+         * ExplorationBackendApiService to ensure the cache is
+         * not out-of-date with any updates made by this backend API service.
+         */
+        updateExploration: function(
+            explorationId, explorationVersion, commitMessage, changeList) {
+          return $q(function(resolve, reject) {
+            _updateExploration(
+              explorationId, explorationVersion, commitMessage, changeList,
+              resolve, reject);
+          });
+        },
+
+        deleteExploration: function(
+          explorationId, role) {
+          return $q(function(resolve, reject) {
+            _deleteExploration(
+              explorationId, role, resolve, reject);
+          });
+        }
+      };
+    }]);
```

Comment:
NB: you'll know something is wrong when this indentation level doesn't match that of the opening braces.